### ISSUE TITLE
BatchNorm example

### ISSUE THREAD

**Issue by unhandyandy** [FluxML/Flux.jl#321]
I tried out the example you give for the BatchNorm and got an error.

```
julia> m = Chain(
         Dense(28^2, 64),
         BatchNorm(64, relu),
         Dense(64, 10),
         BatchNorm(10),
         softmax)
Chain(Dense(784, 64), BatchNorm(64, λ = NNlib.relu), Dense(64, 10), BatchNorm(10), NNlib.softmax)

julia> m(rand(28^2))
ERROR: arraysize: dimension out of range
Stacktrace:
 [1] (::Flux.BatchNorm{NNlib.#relu,TrackedArray{…,Array{Float64,1}},Array{Float64,1},Float64})(::TrackedArray{…,Array{Float64,1}}) at /home/dabrowsa/.julia/v0.6/Flux/src/layers/normalise.jl:116
 [2] mapfoldl_impl(::Base.#identity, ::Flux.##82#83, ::Array{Float64,1}, ::Array{Any,1}, ::Int64) at ./reduce.jl:46
 [3] (::Flux.Chain)(::Array{Float64,1}) at /home/dabrowsa/.julia/v0.6/Flux/src/layers/basic.jl:31

```

**Comment by CarloLucibello** [comment]
Batch norm can be used only in combination with batched inputs, so you should use try with
```julia
julia> m(rand(28^2, 10))
```

**Comment by unhandyandy** [comment]
Thanks.  But then, how do you make a prediction from a single data point?  m(rand(28^2, 1))?

**Comment by unhandyandy** [comment]
Thanks.  But then, how do you make a prediction from a single data point?  m(rand(28^2, 1))?
